crates/wast/src/component/func.rs

```diff
@@ -471,23 +474,46 @@ impl<'a> Parse<'a> for CanonResourceRep<'a> {
     }
 }
 
-/// Information relating to the `thread.spawn` intrinsic.
+/// Information relating to the `thread.spawn_ref` intrinsic.
 #[derive(Debug)]
-pub struct CanonThreadSpawn<'a> {
+pub struct CanonThreadSpawnRef<'a> {
     /// The function type that is being spawned.
     pub ty: Index<'a>,
 }
 
-impl<'a> Parse<'a> for CanonThreadSpawn<'a> {
+impl<'a> Parse<'a> for CanonThreadSpawnRef<'a> {
     fn parse(parser: Parser<'a>) -> Result<Self> {
-        parser.parse::<kw::thread_spawn>()?;
+        parser.parse::<kw::thread_spawn_ref>()?;
 
         Ok(Self {
             ty: parser.parse()?,
         })
     }
 }
 
+/// Information relating to the `thread.spawn_indirect` intrinsic.
+///
+/// This should look quite similar to parsing of `CallIndirect`.
+#[derive(Debug)]
+pub struct CanonThreadSpawnIndirect<'a> {
+    /// The function type that is being spawned.
+    pub ty: Index<'a>,
+    /// The table that this spawn is going to be indexing.
+    pub table: CoreItemRef<'a, kw::table>,
+}
+
+impl<'a> Parse<'a> for CanonThreadSpawnIndirect<'a> {
+    fn parse(parser: Parser<'a>) -> Result<Self> {
+        parser.parse::<kw::thread_spawn_indirect>()?;
+        let ty = parser.parse()?;
+        let table = parser.parens(|parser| {
+            let span = parser.parse::<kw::table>()?.0;
+            parse_trailing_item_ref(kw::table(span), parser)
+        })?;
```

Comment:
Can this be replaced with `let table = parser.parens(|p| p.parse())?;`?

Or let me rephrase after investigating this, this should be replaced with that and I'll update the other places you copied this from since those should all be updated as well.